Tests/SymbolKitTests/SymbolGraph/Symbol/Swift/GenericConstraintTests.swift

```diff
@@ -0,0 +1,89 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import XCTest
+@testable import SymbolKit
+
+class GenericConstraintTests: XCTestCase {
+
+    func testInitializeConformance() throws {
+        let jsonString = """
+        {
+            "kind": "conformance",
+            "lhs": "Self",
+            "rhs": "Hashable",
+        }
+        """
+        let jsonData = jsonString.data(using: .utf8)!
```

Comment:
(Just mentioning this because I thought you might be interested.) This is a weird bit of Swift API design, but since Swift strings can always be represented as UTF8 – this kind of transformation can be spelled like this to avoid the force unwrap:

```
Data(jsonString.utf8)
```

Reply:
We could also do it as a round trip test where we initialize a value, encode it, decode that and compare against the original.

Reply:
Yes that would work too - my idea was just to compare the result of using the two initializers, since neither one of them had any tests before.